File: tbnt-api/app/schemas/work.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime

# Work Settings Schemas
class WorkSettingsBase(BaseModel):
    start_time: str
    end_time: str

class WorkSettingsCreate(WorkSettingsBase):
    pass

class WorkSettingsUpdate(WorkSettingsBase):
    pass

class WorkSettings(WorkSettingsBase):
    id: int
    user_id: int

    class Config:
        from_attributes = True

# Work Item Schemas
class WorkItemBase(BaseModel):
    type: str
    content: str
    status: Optional[str] = "pending"
    percentage: Optional[int] = 0

class WorkItemCreate(WorkItemBase):
    pass

class WorkItemUpdate(BaseModel):
    content: Optional[str] = None
    status: Optional[str] = None
    percentage: Optional[int] = None

class WorkItem(WorkItemBase):
    id: int
    user_id: int
    created_at: datetime

    class Config:
        from_attributes = True

# Work Record Schemas
class WorkRecordBase(BaseModel):
    date: str
    clock_in_time: Optional[str] = None
    clock_out_time: Optional[str] = None

class WorkRecordCreate(BaseModel):
    pass

class WorkRecord(WorkRecordBase):
    id: int
    user_id: int

    class Config:
        from_attributes = True

File: tbnt-web/src/api/work.ts
import request from '@/utils/request'

export interface WorkSettings {
  id: number
  user_id: number
  start_time: string
  end_time: string
}

export interface WorkSettingsUpdate {
  start_time: string
  end_time: string
}

export interface WorkItem {
  id: number
  user_id: number
  type: 'memo' | 'plan' | 'progress'
  content: string
  status: 'pending' | 'done'
  percentage: number
  created_at: string
}

export interface WorkItemCreate {
  type: 'memo' | 'plan' | 'progress'
  content: string
  status?: 'pending' | 'done'
  percentage?: number
}

export interface WorkItemUpdate {
  content?: string
  status?: 'pending' | 'done'
  percentage?: number
}

// Settings
export const getWorkSettings = () => {
  return request.get<WorkSettings>('/work/settings')
}

export const updateWorkSettings = (data: WorkSettingsUpdate) => {
  return request.put<WorkSettings>('/work/settings', data)
}

// Items
export const getWorkItems = (type?: string) => {
  return request.get<WorkItem[]>('/work/items', { params: { type } })
}

export const createWorkItem = (data: WorkItemCreate) => {
  return request.post<WorkItem>('/work/items', data)
}

export const updateWorkItem = (id: number, data: WorkItemUpdate) => {
  return request.put<WorkItem>(`/work/items/${id}`, data)
}

export const deleteWorkItem = (id: number) => {
  return request.delete<WorkItem>(`/work/items/${id}`)
}

export interface WorkRecord {
  id: number
  user_id: number
  clock_in_time: string
  clock_out_time?: string
  date: string
}

export const clockOut = () => {
  return request.post<WorkRecord>('/work/clock-out')
}

export const getWorkRecords = () => {
  return request.get<WorkRecord[]>('/work/records')
}

export const getTodayWorkRecord = () => {
  return request.get<WorkRecord>('/work/records/today')
}
